File: app foodPa/src/app/model/restaurante.ts
export class Restaurante {
  id: number;
    nome: string;
    cnpj: number;
    telefone: number;
    endereco: string;
    email: string;
    senha: string;
    categoria: string;
    
    constructor(){
      this.id = 0;
        this.nome = "";
        this.cnpj = 0;
        this.telefone = 0;
        this.endereco = "";
        this.email = "";
        this.senha = "";  
        this.categoria = "";
    }

}

File: api foodPa/foodpa/foodpa/src/main/java/br/projeto/foodPa/model/Restaurante.java
package br.projeto.foodPa.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 *
 * @author dougl
 */
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString

public class Restaurante {
    private int id;
    private String nome;
    private int cnpj;
    private String endereco;
    private String email;
    private String senha;
    private String telefone;
    private String categoria;
    private String foto;
}